### PR TITLE
Implements API's using ViewSets (V3)

### PR DESCRIPTION
**Pull request by Hamza-Chaudhary65**
- Uses viewsets.viewset for class based API'S
- improves previous naming conventions
- indentation

### CODE REVIEW

File: fintrack/transactions/models.py [hunk 1]
```diff
@@ -4,18 +4,15 @@
 
 
 class Transaction(models.Model):
-    description = models.TextField(blank=True)
```
Comment (ZAINABFATIMA0) [inline fintrack/transactions/models.py]: Why are we leaving this line empty ?

File: fintrack/transactions/transaction_helper.py
Comment (ZAINABFATIMA0) [inline fintrack/transactions/transaction_helper.py:1]: Why are we adding a prefix "transaction" in this helper file's name when we are already in transactions app.

File: fintrack/transactions/v3/views.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    def revert(self, request, pk):
+        transaction = get_object_or_404(
+            Transaction, 
+            pk=pk, 
+            transaction_from_account__holder=request.user
+        )
+
+        if transaction.is_reverted:
+            response = Response(
+                { "message": "Can not revert this transaction"},
```
Comment (ZAINABFATIMA0) [inline fintrack/transactions/v3/views.py]: There is an empty space immediately after the curly bracket

File: fintrack/transactions/v3/views.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        if transaction.is_reverted:
+            response = Response(
+                { "message": "Can not revert this transaction"}, 
+                status= status.HTTP_400_BAD_REQUEST
+            )
+        else:
+            transaction_manager.handle_revert_transaction(transaction, request.data)
+            response = Response(
+                { "message": "Reverted Successfully"}
```
Comment (ZAINABFATIMA0) [inline fintrack/transactions/v3/views.py]: There is an empty space immediately after the curly bracket

File: fintrack/transactions/v3/views.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+from transactions.serializer import TransactionSerializer
+from transactions.transaction_helper import TransactionManager
+
+transaction_manager = TransactionManager()
+
+
+class TransactionViewset(viewsets.ViewSet):
+
+    def list(self, request):
+        transactions = Transaction.objects.filter(transaction_from_account__in=request.user.bank_accounts.all())
```
Comment (ZAINABFATIMA0) [inline fintrack/transactions/v3/views.py]: This line can be divided into two for better readability purpose

**Review by ZAINABFATIMA0 — APPROVED** [review]

File: fintrack/transactions/models.py (hunk 1 above)
Comment (Hamza-Chaudhary65) [reply fintrack/transactions/models.py]: for readability.

File: fintrack/users/v3/views.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from rest_framework import viewsets
+from rest_framework.response import Response
+
+from users.serializer import UserSerializer
+
+
+class RegistrationViewSet(viewsets.ViewSet):
+
+    permission_classes = []
+    def create(self, request):
```
Comment (sheralim012) [inline fintrack/users/v3/views.py]:
```suggestion
    permission_classes = []

        def create(self, request):
```

**Review by sheralim012 — CHANGES_REQUESTED** [review]

**Review by sheralim012 — APPROVED** [review]

**Pull request merged**